include/kernel.h

```diff
@@ -5888,6 +5888,50 @@ int k_thread_runtime_stats_get(k_tid_t thread,
  */
 int k_thread_runtime_stats_all_get(k_thread_runtime_stats_t *stats);
 
+/**
+ * @brief Enable gathering of runtime statistics for specified thread
+ *
+ * This routine enables the gathering of runtime statistics for the specified
+ * thread.
+ *
+ * @param thread ID of thread
+ * @return -EINVAL if invalid thread ID, otherwise 0
+ */
+extern int k_thread_runtime_stats_enable(k_tid_t thread);
+
+/**
+ * @brief Disable gathering of runtime statistics for specified thread
+ *
+ * This routine disables the gathering of runtime statistics for the specified
+ * thread.
+ *
+ * @param thread ID of thread
+ * @return -EINVAL if invalid thread ID, otherwise 0
+ */
+extern int k_thread_runtime_stats_disable(k_tid_t thread);
+
+/**
+ * @brief Enable gathering of CPU runtime statistics on the current CPU
+ *
+ * This routine enables the gathering of CPU runtime statistics on the current
+ * core. Note that it does not affect the gathering of similar statistics for
+ * threads that are running on the core.
+ *
```

Comment:
Still not sure why we want "on the current core" exposed to the API?  What would be bad if we just enforced that this function was global:  "This routine enables the gathering of runtime statistics from threads running on all CPUs".

If there's not some specific problem being solved, maybe we can leave the API (which is a promise that's hard to change) as simple as possible?

Also note some terminology glitchery: obviously "cpu" and "core" end up synonymous in all normal contexts.  Nonetheless I've tried (how successfully is left as an exercise for the reader) to be rigorous in Zephyr about using "cpu" exclusively as far as the general kernel goes.  The word "core" should ideally appear only in the platform layer where it reflects jargon that we can't control.

Reply:
My thinking about "on the current core" was that it may not be guaranteed that each cpu will use the same timer. As a result, to do this properly across all cores increases the complexity (IPIs to inform each core to stop and update its stats timings accordingly).

However, if this is a case that we don't want/need to worry about, then I can update this to start/stop the gathering of cpu statistics based on the cycles read from the one timer source.

Reply:
Ah, OK.  Yeah, timers are guaranteed consistent across cores, that's a hard requirement.  Everything in the timer subsystem breaks if you don't have cross-cpu time accuracy.

Let's take the core stuff out of the API.  It just seems like a needless complication to me.